testplan/runnable/interactive/base.py

```diff
@@ -271,24 +245,36 @@ def run_test_case(
             default local runner.
         :param await_results: Whether to block until the testcase is finished,
             defaults to True.
-        :return: If await_results is True, returns a list of test results.
-            Otherwise, returns an async result object.
+        :return: If await_results is True, returns a testcase report.
+            Otherwise, returns a future which will yield a testcase report when
+            ready.
         """
+        if not await_results:
+            return self._run_async(
+                self.run_test_case, test_uid, suite_uid, case_uid, runner_uid
+            )
+
         test = self.test(test_uid, runner_uid=runner_uid)
         irunner = test.cfg.interactive_runner(test)
         test_run_generator = irunner.run(suite=suite_uid, case=case_uid)
 
-        if await_results:
-            return self._run_all_test_operations(test_run_generator)
-        else:
-            return self._run_async(
-                self._run_all_test_operations, test_run_generator
+        with self._auto_start_stop_environment(test_uid, runner_uid):
+            test_report = self._merge_test_reports(
```

Comment:
why do we need to do the merge there, we just ran one testcase.

Reply:
If the testcase is parametrized, we will run all parametrizations here - which will generate multiple reports.